sealedenumprocessor/src/main/kotlin/com/livefront/sealedenum/internal/SealedEnumProcessor.kt

```diff
@@ -132,23 +136,47 @@ internal class SealedEnumProcessor(
             }
         }
 
+        /**
+         * The [ImmutableKmClass] for the sealed class's companion object.
+         */
+        val sealedClassCompanionObjectKmClass =
+            @Suppress("TooGenericExceptionCaught")
+            try {
+                sealedClassCompanionObjectElement.toImmutableKmClass()
+            } catch (exception: Exception) {
+                if (!isAnnotatedByMetadataOnly) {
+                    printError(ERROR_ELEMENT_IS_NOT_KOTLIN_CLASS, sealedClassCompanionObjectElement)
+                }
+                return null
+            }.apply {
+                if (!isCompanionObject) {
+                    // If this element wasn't explicitly annotated, quietly fail if the class isn't a companion object.
+                    if (!isAnnotatedByMetadataOnly) {
+                        printError(ERROR_ELEMENT_IS_NOT_COMPANION_OBJECT, sealedClassCompanionObjectElement)
```

Comment:
Cool.  I just got this error.

I know you mentioned not having touched the KDoc yet (all in good time), so I ran a scratch file test on this branch with a familiar example,

```
@GenSealedEnum
sealed class MyPet {
  object Fido : MyPet()
  object Cleo : MyPet()
}
```

And it pointed me to the expected generated code, along with this error string.  Nice.

I also played around with the `companion`, noting that a clueless developer can shadow their own generated property, 

```
companion object {
  val values: Int = 0
}
```

Silly? yes.  If this is super easy to detect then maybe worth a warning?  It's basically just a `RTFM` case where we want to reserve some property names without them stepping on it.

Reply:
I added #11 to address this issue. This will be a bit of work, since we will have to check all sealed classes, sealed class subclasses, and sealed objects.